### PR TITLE
Added Support for Sorting in Table

### LINKED ISSUE #428: Support sorting in table header
@edwinbbu _a

Design Link - [Figma File](https://www.figma.com/file/9rPsUXEWN7GA3MNuwd66xC/neetoChat?node-id=665%3A8009)

### PR DESCRIPTION
**Pull request by ajmaln**
- Fixes #428 

@ajmaln _a

### CODE REVIEW

**Comment by ajmaln** [comment]
@edwinbbu _a Please review.

cc: @karthiknmenon @praveen-murali-ind

File: lib/components/Table.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.Head.displayName = "Table.Head";
+Table.Body.displayName = "Table.Body";
+Table.Foot.displayName = "Table.Foot";
+Table.TH.displayName = "Table.TH";
```
Comment (edwinbbu) [inline lib/components/Table.js:168]: Can we rename to `Table.Th`, `Table.Tr`, `Table.Td`. If you can find a better name, then we can have that. Consistency is missing from the previous name.

File: example/src/Pages/Layouts.js [hunk 2]
```diff
@@ -8,34 +15,46 @@ import {
...
-      fill="none"
-      viewBox="0 0 24 24"
-    >
-      <path stroke={color} strokeWidth="1.5" d="M3 7.25L21 7.25"></path>
-      <path stroke={color} strokeWidth="1.5" d="M3 11.25L15 11.25"></path>
-      <path stroke={color} strokeWidth="1.5" d="M3 15.25L11 15.25"></path>
-    </svg>
-  );
-}
+const useSortTable = (items, config = null) => {
```
Comment (edwinbbu) [inline example/src/Pages/Layouts.js:26]: Can we export useSortTable from the table component? We do not need to repeat this code in the all the applications.

**Comment by ajmaln** [comment]
@edwinbbu _a Please review and let me know if any further changes need to be done.

**Comment by karthiknmenon** [comment]
Closing wrt https://github.com/bigbinary/neeto-ui/pull/597.

**Pull request closed without merging**